echoshader/dev_class/echo_map.py

```diff
@@ -0,0 +1,436 @@
+def plot_track(MVBS_ds,
+               opts_line = geoviews.opts(width = 600, height = 400, \
+               color = 'red',tools = ['hover'], line_width = 1),
+               opts_point = geoviews.opts(color = 'red',\
+               tools = ['hover'], size = 8),
+               map_tiles= 'Wikipedia'):
+
+        '''
+
+        Parameters
+        ----------
+        MVBS_ds : xarray.Dataset
+            MVBS Dataset with coordinates 'ping_time', 'channel', 'echo_range'
+            MVBS Dataset has been combined with longitude & latitude coordinates using echopype
+
+        opts_line : geoviews.opts, default : geoviews.opts(width = 600, height = 400, \
+               color = 'red',tools = ['hover'], line_width = 1)
+            Modify the style of line
+
+        opts_point : geoviews.opts, default : geoviews.opts(color = 'red',\
+               tools = ['hover'], size = 8)
+            Modify the style of point
+
+        map_tiles : str, default : 'Wikipedia'
+            See more in : https://geoviews.org/gallery/bokeh/tile_sources.html
+
+        Returns
+        -------
+        ship_track * start_node * tiles ：holoviews.Overlay
+            Combined chart( track, start point and map tile )
+
+        '''
+
+    # convert xarray data to geoviews data
+    all_pd_data = pandas.concat([
+        pandas.DataFrame(MVBS_ds.longitude.values, columns=['Longitude']),
+        pandas.DataFrame(MVBS_ds.latitude.values, columns=['Latitude']),
+        pandas.DataFrame(MVBS_ds.ping_time.values, columns=['Ping Time'])
+    ],
+                                axis=1)
+
+    all_pd_data = all_pd_data.dropna(axis=0, how='any')
+
+    # get map tiles
+    tiles = getattr(geoviews.tile_sources, map_tiles)
+
+    # plot path
+    ship_track = geoviews.Path([all_pd_data],
+                               kdims=['Longitude', 'Latitude'],
+                               vdims=['Ping Time','Longitude', 'Latitude'])\
+                               .opts(opts_line)
+
+    # plot start node
+    start = all_pd_data.iloc[0].values.tolist()
+
+    start_node = geoviews.Points([start],
+                                  kdims=['Longitude', 'Latitude'],)\
+                                  .opts(opts_point)
+
+    return ship_track * start_node * tiles
+
+
+class EchoMap(echo_gram.Echogram):
+
+    """
+    A class for plotting basic echogram
+
+    Attributes
+    ----------
+    MVBS_ds : xarray.Dataset
+        MVBS Dataset with coordinates 'ping_time', 'channel', 'echo_range'
+
+    plot_types : str, default : 'image'
+        Usually using 'image' or 'quadmesh'
+        See 'image' in : https://hvplot.holoviz.org/reference/xarray/image.html#xarray-gallery-image
+        See 'quadmesh' in : https://hvplot.holoviz.org/reference/xarray/quadmesh.html#xarray-gallery-quadmesh
+        Notice : When using 'quadmesh', box_select or some tools may not be applicable
+
+    datetime_range_input_model : bool, default : 'True'
```

Comment:
Class attributes do not need default values... Please remove all default in Classes.